crates/time_format/src/time_format.rs

```diff
@@ -0,0 +1,303 @@
+            "{:02}/{:02}/{}",
+            timestamp_local_date.day(),
+            timestamp_local_date.month() as u32,
+            timestamp_local_date.year()
+        ),
+    }
+}
+
+/// Returns true if the locale is recognized as a 12-hour time locale.
+fn is_12_hour_time_by_locale(locale: &str) -> bool {
+    [
+        "es-MX", "es-CO", "es-SV", "es-NI",
+        "es-HN", // Mexico, Colombia, El Salvador, Nicaragua, Honduras
+        "en-US", "en-CA", "en-AU", "en-NZ", // U.S, Canada, Australia, New Zealand
+        "ar-SA", "ar-EG", "ar-JO", // Saudi Arabia, Egypt, Jordan
+        "en-IN", "hi-IN", // India, Hindu
+        "en-PK", "ur-PK", // Pakistan, Urdu
+        "en-PH", "fil-PH", // Philippines, Filipino
+        "bn-BD", "ccp-BD", // Bangladesh, Chakma
+        "en-IE", "ga-IE", // Ireland, Irish
+        "en-MY", "ms-MY", // Malaysia, Malay
+    ]
+    .contains(&locale)
+}
+
+#[cfg(target_os = "macos")]
+mod macos {
+    use super::*;
+    use core_foundation::base::TCFType;
+    use core_foundation::date::CFAbsoluteTime;
+    use core_foundation::string::CFString;
+    use core_foundation_sys::date_formatter::CFDateFormatterCreateStringWithAbsoluteTime;
+    use core_foundation_sys::date_formatter::CFDateFormatterRef;
+    use core_foundation_sys::locale::CFLocaleRef;
+    use core_foundation_sys::{
+        base::kCFAllocatorDefault,
+        date_formatter::{
+            kCFDateFormatterNoStyle, kCFDateFormatterShortStyle, CFDateFormatterCreate,
+        },
+        locale::CFLocaleCopyCurrent,
+    };
+
+    pub fn format_time(timestamp: &time::OffsetDateTime) -> Result<String> {
+        format_with_date_formatter(timestamp, TIME_FORMATTER.with(|f| *f))
+    }
+
+    pub fn format_date(timestamp: &time::OffsetDateTime) -> Result<String> {
+        format_with_date_formatter(timestamp, DATE_FORMATTER.with(|f| *f))
+    }
+
+    fn format_with_date_formatter(
+        timestamp: &time::OffsetDateTime,
+        fmt: CFDateFormatterRef,
+    ) -> Result<String> {
+        const UNIX_TO_CF_ABSOLUTE_TIME_OFFSET: i64 = 978307200;
+        // Convert timestamp to macOS absolute time
+        let timestamp_macos = timestamp.unix_timestamp() - UNIX_TO_CF_ABSOLUTE_TIME_OFFSET;
+        let cf_absolute_time = timestamp_macos as CFAbsoluteTime;
+        unsafe {
+            let s = CFDateFormatterCreateStringWithAbsoluteTime(
+                kCFAllocatorDefault,
+                fmt,
+                cf_absolute_time,
+            );
+            Ok(CFString::wrap_under_create_rule(s).to_string())
+        }
+    }
+
+    thread_local! {
+        static CURRENT_LOCALE: CFLocaleRef = unsafe { CFLocaleCopyCurrent() };
+        static TIME_FORMATTER: CFDateFormatterRef = unsafe {
+            CFDateFormatterCreate(
+                kCFAllocatorDefault,
+                CURRENT_LOCALE.with(|locale| *locale),
+                kCFDateFormatterNoStyle,
+                kCFDateFormatterShortStyle,
+            )
+        };
```

Comment:
As these globals live for the whole lifetime of the application, the memory will be cleaned up by the OS if zed is shutdown. So we do not need any calls to `CFRelease` here, is that correct?